src/ResourceManagement/Logic/Logic.Tests/InMemoryTests/IntegrationAccountSessionInMemoryTests.cs

```diff
@@ -0,0 +1,316 @@
+            Assert.Throws<CloudException>(() => client.Sessions.ListByIntegrationAccountsNext(nextPageLink: Constants.NextPageLink));
+        }
+
+        [Fact]
+        public void IntegrationAccountSession_ListByResourceGroupNext_Success()
+        {
+            var handler = new RecordedDelegatingHandler();
+            var client = this.CreateIntegrationAccountClient(handler);
+
+            handler.Response = new HttpResponseMessage
+            {
+                StatusCode = HttpStatusCode.OK,
+                Content = this.SessionList
+            };
+
+            var result = client.Sessions.ListByIntegrationAccountsNext(nextPageLink: Constants.NextPageLink);
+
+            handler.Request.ValidateAuthorizationHeader();
+            handler.Request.ValidateMethod(HttpMethod.Get);
+
+            this.ValidateSessionList(result);
+        }
+
+        [Fact]
+        public void IntegrationAccountSession_CreateOrUpdate_Exception()
+        {
+            var handler = new RecordedDelegatingHandler();
+            var client = this.CreateIntegrationAccountClient(handler);
+
+            handler.Response = new HttpResponseMessage
+            {
+                StatusCode = HttpStatusCode.InternalServerError,
+                Content = new StringContent(string.Empty)
+            };
+
+            Assert.Throws<ValidationException>(() => client.Sessions.CreateOrUpdate(
+                resourceGroupName: null,
+                integrationAccountName: "IntegrationAccountName",
+                sessionName: "SessionName",
+                session: new IntegrationAccountSession()));
+            Assert.Throws<ValidationException>(() => client.Sessions.CreateOrUpdate(
+                resourceGroupName: ResourceGroupName,
+                integrationAccountName: null,
+                sessionName: "SessionName",
+                session: new IntegrationAccountSession()));
+            Assert.Throws<ValidationException>(() => client.Sessions.CreateOrUpdate(
+                resourceGroupName: ResourceGroupName,
+                integrationAccountName: "IntegrationAccountName",
+                sessionName: null,
+                session: new IntegrationAccountSession()));
+            Assert.Throws<ValidationException>(() => client.Sessions.CreateOrUpdate(
+                resourceGroupName: ResourceGroupName,
+                integrationAccountName: "IntegrationAccountName",
+                sessionName: "SessionName",
+                session: null));
+            Assert.Throws<CloudException>(() => client.Sessions.CreateOrUpdate(
+                resourceGroupName: ResourceGroupName,
+                integrationAccountName: "IntegrationAccountName",
+                sessionName: "SessionName",
+                session: new IntegrationAccountSession()));
+        }
+
+        [Fact]
+        public void IntegrationAccountSession_CreateOrUpdate_OK()
+        {
+            var handler = new RecordedDelegatingHandler();
```

Comment:
This comment is somewhat similar to the previous one.

You are correct that the most interesting tests here are the negative test cases. The positive tests do not add on top of the recorded tests, they are more a step in the development process as they can be written and run prior to touching the service at all. These are consistent with the approach for the remaining artifacts in this RP. I do not see a harm in them.